packaging/os/rpm_key.py

```diff
@@ -115,18 +114,18 @@ def __init__(self, module):
 
     def fetch_key(self, url):
         """Downloads a key from url, returns a valid path to a gpg key"""
-        try:
-            rsp, info = fetch_url(self.module, url)
-            key = rsp.read()
-            if not is_pubkey(key):
-                self.module.fail_json(msg="Not a public key: %s" % url)
-            tmpfd, tmpname = tempfile.mkstemp()
-            tmpfile = os.fdopen(tmpfd, "w+b")
-            tmpfile.write(key)
-            tmpfile.close()
-            return tmpname
-        except urllib2.URLError, e:
-            self.module.fail_json(msg=str(e))
+        rsp, info = fetch_url(self.module, url)
+        if info['status'] != 200:
+            self.module.fail_json(msg="failed to fetch key at %s , error was: %s" % (url, info['msg']))
```

Comment:
IIRC, urllib2 follow redirections by default. I was unsure on this as well, and did look at others modules to see what they did. For example:

https://github.com/ansible/ansible-modules-core/blob/devel/system/authorized_key.py#L380

https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/apt_repository.py#L340

And my understanding of the code of fetch_url is that it set the code to be 200 if all is well:

https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L764

So it should be ok (but maybe I am wrong, and that's why we have review)